src/main/java/org/wso2/carbon/module/swiftiso20022/mt/parsers/MTParser.java

```diff
@@ -0,0 +1,247 @@
+
+                case Field115.TAG:
+                    userHeaderBlock.setAddresseeInformation(Field115.parse(fields.get(key)));
+                    break;
+
+                case Field119.TAG:
+                    userHeaderBlock.setValidationFlag(Field119.parse(fields.get(key)));
+                    break;
+
+                case Field121.TAG:
+                    userHeaderBlock.setEndToEndReference(Field121.parse(fields.get(key)));
+                    break;
+
+                case Field165.TAG:
+                    userHeaderBlock.setPaymentReleaseInformation(Field165.parse(fields.get(key)));
+                    break;
+
+                case Field423.TAG:
+                    userHeaderBlock.setBalanceCheckpoint(Field423.parse(fields.get(key)));
+                    break;
+
+                case Field424.TAG:
+                    userHeaderBlock.setRelatedReference(Field424.parse(fields.get(key)));
+                    break;
+
+                case Field433.TAG:
+                    userHeaderBlock.setSanctionsScreeningInformation(Field433.parse(fields.get(key)));
+                    break;
+
+                case Field434.TAG:
+                    userHeaderBlock.setPaymentControlsInformation(Field434.parse(fields.get(key)));
+                    break;
+
+                // If tag didn't match any of the above tags, it is not allowed
+                default:
+
+                    String errorMessage = String.format(ConnectorConstants.ERROR_FIELD_NOT_ALLOWED_IN_BLOCK,
+                            key, ConnectorConstants.USER_HEADER_BLOCK);
+                    log.error(errorMessage);
+                    throw new MTMessageParsingException(errorMessage);
+            }
+        }
+
+        return userHeaderBlock;
+    }
+
+    /**
+     * Method to parse user header block and return UserHeaderBlock object.
+     *
+     * @param trailerBlockString String containing trailer block fields
+     *                           example -> {CHK:123456789ABC}{DLM:}
+     * @return A {@link TrailerBlock} model with values assigned by parsing trailer block
+     * @throws MTMessageParsingException if the string cannot be parsed.
+     */
+    private static TrailerBlock parseTrailerBlock(String trailerBlockString) throws MTMessageParsingException {
+
+        if (trailerBlockString == null) {
+            return null;
+        }
+
+        // extract fields as a map of tag and value
+        Map<String, String> fields = MTParserUtils.extractFieldsWithinCurlyBrackets(trailerBlockString);
+
+        if (fields.isEmpty()) {
+
+            String errorMessage = String.format(
+                    ConnectorConstants.ERROR_BLOCK_INVALID_FORMAT, ConnectorConstants.TRAILER_BLOCK);
+            log.error(errorMessage);
+            throw new MTMessageParsingException(errorMessage);
+        }
+
+        TrailerBlock trailerBlock = new TrailerBlock();
+
+        for (String key : fields.keySet()) {
+
+            switch (key) {
```

Comment:
Same comment https://github.com/wso2-extensions/mediation-swift-iso20022-module/pull/11/files#r1574133144

Reply:
Removed in a82da3f30afc3995c983ec8d68a04acfc363b6c0